src/main/kotlin/io.javalin/HaltException.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin
+
+import javax.servlet.http.HttpServletResponse
+
+class HaltException : RuntimeException {
+    var statusCode = HttpServletResponse.SC_OK
+    var body = "Execution halted"
+
+    constructor() {}
+
+    constructor(statusCode: Int) {
```

Comment:
if you are calling both variations from java then yes you have to do some extra contortions. one can minimize this some by adding the `@JvmOverloads` annotation which will auto-generate telescoping methods for java, but you will still have to keep the body constructor if you need to support that one.